### PR TITLE
[BUG FIX] Segmentation sample u2netp model results

### PR DESCRIPTION
**Pull request by gursimarsingh**
PR resloves #25753 related to incorrect output from u2netp model in segmentation sample

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: samples/dnn/segmentation.cpp [hunk 1]
```diff
@@ -146,29 +146,21 @@ int main(int argc, char **argv)
...
-            foreground_overlay = Mat::zeros(frame.size(), frame.type());
-            background_overlay = Mat::zeros(frame.size(), frame.type());
-            // Set foreground (object) to red and background to blue
-            foreground_overlay.setTo(Scalar(0, 0, 255), thresholded_mask);
-            Mat inverted_mask;
-            bitwise_not(thresholded_mask, inverted_mask);
-            background_overlay.setTo(Scalar(255, 0, 0), inverted_mask);
-            // Blend the overlays with the original frame
-            addWeighted(frame, 1, foreground_overlay, 0.5, 0, foreground_segmented);
-            addWeighted(foreground_segmented, 1, background_overlay, 0.5, 0, frame);
```
Comment (asmorkalov) [inline samples/dnn/segmentation.cpp:168]: Why do you remove the solution with overlays? It looks better, IMHO.

File: samples/dnn/models.yml [hunk 2]
```diff
@@ -256,9 +256,9 @@ u2netp:
     url: "https://github.com/danielgatis/rembg/releases/download/v0.0.0/u2netp.onnx"
     sha1: "0a99236f0d5c1916a99a8c401b23e5ef32038606"
   model: "u2netp.onnx"
-  mean: [123.6, 116.2, 103.5]
+  mean: [0.485, 0.456, 0.406]
   scale: 0.019
   width: 320
   height: 320
-  rgb: true
```
Comment (asmorkalov) [inline samples/dnn/models.yml:263]: It's incorrect change. I checked the rembg/bg.py sample in the model release and the model expects RGB input, but not BGR.

File: samples/dnn/models.yml [hunk 3]
```diff
@@ -256,9 +256,9 @@ u2netp:
     url: "https://github.com/danielgatis/rembg/releases/download/v0.0.0/u2netp.onnx"
     sha1: "0a99236f0d5c1916a99a8c401b23e5ef32038606"
   model: "u2netp.onnx"
-  mean: [123.6, 116.2, 103.5]
```
Comment (asmorkalov) [inline samples/dnn/models.yml:259]: The change looks suspicious. OpenCV subtracts mean and then divides by scale. Input image has 0..255 range.

File: samples/dnn/segmentation.cpp (hunk 1 above)
Comment (gursimarsingh) [reply samples/dnn/segmentation.cpp]: @asmorkalov The models.yml preprocessing pararmeters have been fixed. For overlay I have added the solution back but without thresholding. The model does very well in fine areas, the effect is lost with thresholding. This is how it looks now

hockey
horse

File: samples/dnn/segmentation.cpp [hunk 4]
```diff
@@ -146,29 +146,35 @@ int main(int argc, char **argv)
...
             // Set foreground (object) to red and background to blue
-            foreground_overlay.setTo(Scalar(0, 0, 255), thresholded_mask);
-            Mat inverted_mask;
-            bitwise_not(thresholded_mask, inverted_mask);
-            background_overlay.setTo(Scalar(255, 0, 0), inverted_mask);
+            std::vector<Mat> channels;
+            split(foreground_overlay, channels);
+            channels[2] = mask;  // Red foreground
+            merge(channels, foreground_overlay);
+
```
Comment (asmorkalov) [inline samples/dnn/segmentation.cpp:169]: split is not needed here. You can create single channel Mat with zeros and then use it several times for merge, e.g.:
```
Mat all_zeros = Mat::zeros(frame.size(), CV_8UC1);
vector<Mat> channels = {all_zeros, all_zeros, mask};
merge(channels, foreground_overlay);
```

**Review by asmorkalov — APPROVED** [review]
Looks good to me, but some simplification ma be applied.

**Review by asmorkalov — APPROVED** [review]

**Pull request merged**